doc/conf.py

```diff
@@ -530,3 +529,80 @@ def setup(app):
     else:
         bld_type = 'rel'
     app.add_config_value('releaselevel', bld_type, 'env')
+
+# -----------------------------------------------------------------------------
+# Source code links
+# -----------------------------------------------------------------------------
+link_github = True
+# You can add viewcode extension to build old way with link_github = False
+# extensions.append('sphinx.ext.viewcode')
+if link_github:
+    import re
+    import inspect
+    from os.path import relpath, dirname
+    for name in ['sphinx.ext.linkcode', 'linkcode', 'numpydoc.linkcode']:
```

Comment:
`sphinx.ext.linkcode` was added in Sphinx 1.2, and we require 1.8.1, so I don't know why you need all the fallbacks.